ext/span.c

```diff
@@ -560,6 +561,7 @@ void ddtrace_drop_span(ddtrace_span_data *span) {
 
     if (stack->root_span == span) {
         ddtrace_switch_span_stack(stack->parent_stack);
+        stack->root_span = NULL;
```

Comment:
I assume these changes in `ext/span.c` are related to the failures in the randomized tests you found?

Reply:
Yes :-)